File: tailadmin/src/types/order.ts
export interface OrderItem {
  id: number;
  order_id: number;
  product_id: number;
  product_name: string;
  quantity: number;
  price: number;
  discount_type: string;
  discount_value: number;
  subtotal: number;
}

export interface Order {
  id: number;
  invoice_number: string;
  cashier_id: number;
  customer_name: string;
  status: 'pending' | 'paid' | 'cancelled';
  subtotal: number;
  discount_total: number;
  voucher_code: string;
  voucher_discount: number;
  tax_total: number;
  grand_total: number;
  payment_method: 'cash' | 'debit' | 'qris';
  payment_amount: number;
  change_amount: number;
  created_at: string;
  cashier?: { id: number; full_name: string };
  items?: OrderItem[];
}

export interface CartItem {
  product_id: number;
  variant_id?: number;
  variant_name?: string;
  product_name: string;
  price: number;
  quantity: number;
  stock: number;
  product_image?: string;
}

export interface CreateOrderInput {
  items: { product_id: number; variant_id?: number; quantity: number }[];
  voucher_code?: string;
  customer_name?: string;
  payment_method: string;
  payment_amount: number;
}

File: backend/internal/models/order.go
package models

import "time"

type Order struct {
	ID             uint        `gorm:"primaryKey" json:"id"`
	InvoiceNumber  string      `gorm:"unique;not null;size:30" json:"invoice_number"`
	CashierID      uint        `gorm:"not null" json:"cashier_id"`
	CustomerName   string      `gorm:"size:100" json:"customer_name"`
	Status         string      `gorm:"not null;default:pending;size:20" json:"status"`
	Subtotal       int64       `gorm:"not null" json:"subtotal"`
	DiscountTotal  int64       `gorm:"default:0" json:"discount_total"`
	VoucherCode    string      `gorm:"size:50" json:"voucher_code"`
	VoucherDiscount int64      `gorm:"default:0" json:"voucher_discount"`
	TaxTotal       int64       `gorm:"default:0" json:"tax_total"`
	GrandTotal     int64       `gorm:"not null" json:"grand_total"`
	PaymentMethod  string      `gorm:"size:20" json:"payment_method"`
	PaymentAmount  int64       `gorm:"default:0" json:"payment_amount"`
	ChangeAmount   int64       `gorm:"default:0" json:"change_amount"`
	CreatedAt      time.Time   `json:"created_at"`
	UpdatedAt      time.Time   `json:"updated_at"`
	Cashier        User        `gorm:"foreignKey:CashierID" json:"cashier,omitempty"`
	Items          []OrderItem `gorm:"foreignKey:OrderID" json:"items,omitempty"`
}
